### PR TITLE
SQL: Fix elasticsearch-sql-cli under docker

### LINKED ISSUE #57744: elasticsearch-sql-cli: ERROR: E is not a recognized option with docker.elastic.co/elasticsearch/elasticsearch:7.6.2 distribution
**Elasticsearch version** (`bin/elasticsearch --version`):
```
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
Version: 7.6.2, Build: default/docker/ef48eb35cf30adf4db14086e8aabd07ef6fb113f/2020-03-26T06:34:37.794943Z, JVM: 13.0.2
```
**Plugins installed**: []
N/A
**JVM version** (`java -version`):
```
$ /usr/share/elasticsearch/jdk/bin/java
openjdk version "13.0.2" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.2+8)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.2+8, mixed mode, sharing)
```
**OS version** (`uname -a` if on a Unix-like system):
```
Linux b6e0e6b3e8a9 5.4.7-050407-generic #201912311234 SMP Tue Dec 31 17:38:50 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
**Description of the problem including expected versus actual behavior**:

I'm using the docker image `docker.elastic.co/elasticsearch/elasticsearch:7.6.2`.

The elasticsearch-sql-cli command included in the image doesn't seem to work:

```
$ ./bin/elasticsearch-sql-cli http://localhost:9200
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
Elasticsearch SQL CLI

Non-option arguments:
uri

Option                   Description
------                   -----------
-b, --binary <Boolean>   Disable binary communication. Enabled by default.
                           Accepts 'true' or 'false' values. (default: true)
-c, --check <Boolean>    Enable initial connection check on startup (default:
                           true)
-d, --debug              Enable debug logging
-h, --help               Show help
-k, --keystore_location  Location of a keystore to use when setting up SSL. If
                           specified then the CLI will prompt for a keystore
                           password. If specified when the uri isn't https then
                           an error is thrown.
-s, --silent             Show minimal output
-v, --verbose            Show verbose output
ERROR: E is not a recognized option
``` 
(run using docker exec -it)

I didn't find bug reports for this error with elasticsearch-sql-cli, but I found this one: https://github.com/elastic/elasticsearch/issues/19931

The issue mentions this can happen if one fails to provide an argument to the command. In my case I'm providing the only required argument.  

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem,
including (e.g.) index creation, mappings, settings, query etc.  The easier
you make for us to reproduce it, the more likely that somebody will take the
time to look at it.

 1. start the elasticsearch container(e.g. `container=$(docker run --rm -it -d docker.elastic.co/elasticsearch/elasticsearch:7.6.2)`
 2. try and run the elasticsearch-sql-cli command inside the container(e.g. `docker exec -it $container ./bin/elasticsearch-sql-cli http://localhost:9200`)

### PR DESCRIPTION
**Pull request by palesz**
Calling `./bin/elasticsearch-sql-cli http://localhost:9200`under
docker failed with `ERROR: E is not a recognized option` error
message.

The root cause is that under docker the environmental variables are
automatically appended to the arguments of the sql cli. `"$@"` will
contain `-Evar=value` strings added by the `elasticsearch-env`
(line 122). This change uses the original argument list (saves it
before it is modified) when the sql cli is called.

Fixes #57744

### CODE REVIEW

**Review by matriv — APPROVED** [review]

File: x-pack/plugin/sql/src/main/bin/elasticsearch-sql-cli [hunk 1]
```diff
@@ -4,6 +4,8 @@
 # or more contributor license agreements. Licensed under the Elastic License;
 # you may not use this file except in compliance with the Elastic License.
 
+CLI_PARAMETERS="$@"
```
Comment (costin) [inline x-pack/plugin/sql/src/main/bin/elasticsearch-sql-cli:7]: Please ping the core-infra folks as well. My preference would be for `CLI_PARAMETERS` to follow CLI_JAR declaration to maintain the structure of the script.

**Review by costin — APPROVED** [review]
Left some comments

File: x-pack/plugin/sql/src/main/bin/elasticsearch-sql-cli (hunk 1 above)
Comment (palesz) [reply x-pack/plugin/sql/src/main/bin/elasticsearch-sql-cli:7]: `elasticsearch-env` appends to `"$@"`, so it has to be before the `source .../elasticsearch-env` line. This is where the bug comes from.

**Comment by costin** [comment]
@jakelandis could somebody from your team take a quick look at this PR? Thanks!

**Review by bpintea — APPROVED** [review]

**Review by astefan — APPROVED** [review]
LGTM if Infra is ok with it.

**Review by pugnascotia — CHANGES_REQUESTED** [review]
The script change looks fine, but let's add a test to the `DockerTests` class as well. See `DockerTests.test020PluginsListWithNoPlugins` for inspiration - we probably just need to execute `./bin/elasticsearch-sql-cli --help` and check that it doesn't explode.

**Comment by palesz** [comment]
> The script change looks fine, but let's add a test to the `DockerTests` class as well. See `DockerTests.test020PluginsListWithNoPlugins` for inspiration - we probably just need to execute `./bin/elasticsearch-sql-cli --help` and check that it doesn't explode.

@pugnascotia What about instead of adding a new test to the DockerTests, we just remove this `assumeFalse(isDocker)` from the SqlCliTest? https://github.com/elastic/elasticsearch/blob/990381fb1ead11e07227cd329a7ff1f3fd3fbed1/qa/os/src/test/java/org/elasticsearch/packaging/test/SqlCliTests.java#L34

**Comment by pugnascotia** [comment]
@palesz interesting, I didn't know that existed. That approach would be better, and I believe `PackagingTestCase` does have Docker support now (it didn't use to). If we do run into problems, we'll fix them.

**Comment by palesz** [comment]
The `SqlCliTests` ran successfully locally using the `./gradlew :qa:os:destructiveDistroTest.default-docker --tests "org.elasticsearch.packaging.test.SqlCliTests"` after the script fix and enabling the test execution for docker packaging.

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**